toolkit/tools/internal/safemount/safemount.go

```diff
@@ -71,23 +68,48 @@ func (m *Mount) Target() string {
 	return m.target
 }
 
+// Close removes the system mount and fails if the device is still busy.
+// CleanClose and Close are safe to call multiple times.
+func (m *Mount) CleanClose() error {
+	return m.close(false /*async*/)
+}
+
 // Close removes the system mount.
-// This function is safe to call multiple times.
-func (m *Mount) Close() error {
-	var err error
+// The unmount is performed asynchronously. This reduces the likelihood of the unmount failing
+// (thus ensuring the user's system is left in a clean state). But it doesn't provide any
+// guarantees about the validity of the written bits.
+// CleanClose and Close are safe to call multiple times.
+func (m *Mount) Close() {
+	err := m.close(true /*async*/)
+	if err != nil {
+		logger.Log.Warnf("%s", err)
+	}
+}
 
-	logger.Log.Debugf("Unmounting (%s)", m.target)
+func (m *Mount) close(async bool) error {
+	var err error
 
 	if m.isMounted {
-		err = unix.Unmount(m.target, 0)
-		if err != nil {
-			return fmt.Errorf("failed to unmount (%s):\n%w", m.target, err)
+		if !async {
```

Comment:
Any concern with asynchronous calls from different threads? Or outside scope?

Reply:
The MIC codebase is all serial at the moment. So, it is not an issue that this class is not thread safe.

What do you mean by "outside scope"?